core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -114,13 +114,13 @@ class Partition(val topic: String,
   def isUnderReplicated: Boolean =
     isLeaderReplicaLocal && inSyncReplicas.size < assignedReplicas.size
 
-  def getOrCreateReplica(replicaId: Int = localBrokerId): Replica = {
+  def getOrCreateReplica(replicaId: Int = localBrokerId, isNew: Boolean = false): Replica = {
     assignedReplicaMap.getAndMaybePut(replicaId, {
       if (isReplicaLocal(replicaId)) {
         val config = LogConfig.fromProps(logManager.defaultConfig.originals,
                                          AdminUtils.fetchEntityConfig(zkUtils, ConfigType.Topic, topic))
-        val log = logManager.createLog(topicPartition, config)
-        val checkpoint = replicaManager.highWatermarkCheckpoints(log.dir.getParentFile.getAbsolutePath)
+        val log = logManager.createLog(topicPartition, config, isNew)
+        val checkpoint = replicaManager.highWatermarkCheckpoints(log.dir.getParent)
```

Comment:
Yes, it is intended. I think `dir.getParent` is exactly the same as `dir.getParentFile.getAbsolutePath` according to their Java doc. `dir.getParent` is a bit shorter. I think parent directories on different disk must have the different name since they are specified as a list of strings using `log.dirs` config. They must be different so that Kafka can distinguish between them.